provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -126,50 +107,84 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
   }
 
   /** Given a job part, create the dataframe to subsequently be used to read/insert data into Vector */
-  private def vectorDF(part: JobPart): DataFrame = {
+  private def getVectorDF(part: JobPart): DataFrame = {
     val rel = VectorRelation(part.column_infos.map(_.toColumnMetadata), part.conf, sqlContext)
     sqlContext.baseRelationToDataFrame(rel)
   }
 
-  /** Helper function to register a dataframe as a temp table with a given `prefix` */
-  private def register(prefix: String, df: DataFrame): String = {
-    val ret = s"${prefix}_${id.incrementAndGet}"
-    df.registerTempTable(ret)
-    ret
-  }
+  /** Given a job part, retrieve its options, if any, or else an empty Map */
+  private def getOptions(part: JobPart): Map[String, String] = part.options.getOrElse(Map.empty[String, String])
 
   /**
-   * Given a job part of "scan" type, generate the SparkSQL statement that can be used to retrieve data from Vector
-   *
-   * @param format the format of the external data source
+   * Given a job part, return the format of the external table, according to the following logic:
+   * 1) return the explicit part.format, if specified
+   * 2) return the extension of part.external_reference, if recognized
+   * 3) throw IllegalArgumentException otherwise
    */
-  private def selectStatement(format: String, part: JobPart): String = {
-    val options = part.options.getOrElse(Map.empty[String, String])
+  private def getFormat(part: JobPart): String =
+    part.format.orElse(PartialFunction.condOpt[String, String](part.external_reference.split("\\.").last) {
+      _ match {
+        case "csv" => "csv"
+        case "parquet" => "parquet"
+        case "json" => "json"
+        case "orc" => "orc"
+        case "avro" => "avro"
+      }
+    }).getOrElse {
+      throw new IllegalArgumentException(s"""Could not derive format of external reference ${part.external_reference},
```

Comment:
do you need the s""" ... """ ? make it s"..." if it doesn't bring nothing extra, it consistent with the rest of str formatted messages